pycalphad/model.py

```diff
@@ -768,9 +768,25 @@ def redlich_kister_sum(self, phase, param_search, param_query):
                     mixing_term *= Mul(*comp_symbols)
                 # is this a higher-order interaction parameter?
                 if len(comps) == 2 and param['parameter_order'] > 0:
-                    # interacting sublattice, add the interaction polynomial
-                    mixing_term *= Pow(comp_symbols[0] - \
-                        comp_symbols[1], param['parameter_order'])
+                    subl_indices, comps_list = list(zip(*[[idx,c] for idx,c in enumerate(param['constituent_array']) if len(c) == 2]))
+                    if len(comps_list) > 2:
+                        raise ValueError("Composition dependent reciprocal parameter is not supported for more than 2 sublattices")
+                    elif len(comps_list) > 1:
+                        if param['parameter_order'] > 2:
+                            raise ValueError("Order of reciprocal parameter is limited to 0, 1 or 2")
+                        # only add (yA - yB) if the sublattice corresponds to the parameter order
+                        # this is based off the equation from HL Lukas, SG Fries, B Sundman
+                        #   Computational Thermodynamics - The Calphad Method, 
+                        #   Ch5 Models for the Gibbs Free Energy, Section 5.8.1. Reciprocal solutions
+                        #   Equation 5.99
+                        # Testing with thermo-calc seems to be that L1 corresponds to the second
+                        # sublattice and L2 corresponds to the first sublattice
+                        if (1-subl_indices.index(subl_index)) == param['parameter_order'] - 1:
+                            mixing_term *= (comp_symbols[0] - comp_symbols[1])
+                    else:
```

Comment:
My reading of 5.8.1.1 is that this would also apply to the case of more than 3 sublattices, can we test that in TC?

Reply:
My interpretation from the last two paragraphs would be for a model with 3+ sublattices, the reciprocal parameter is only defined for two composition dependent terms. So:
- $L_{AB:C:EF} = ^0 L_{AB:C:EF} + (y^0_A - y^0_B)* ^1 L_{AB:C:EF} + (y^2_E - y^2_F)* ^2 L_{AB:C:EF}$ 
- $L_{AB:CD:EF} = ^0 L_{AB:CD:EF}$ - composition dependent reciprocal parameter not supported since there's 3 composition dependent terms

Is this how you interpret it as well?

Now, I suppose equation 5.99 can be expanded to more sublattices as:
$$L_{AB:CD:EF} = ^0 L_{AB:CD:EF} + (y^0_A - y^0_B)* ^1 L_{AB:CD:EF} + (y^1_C - y^1_D)* ^2 L_{AB:CD:EF} + (y^2_E - y^2_F)* ^3 L_{AB:CD:EF}$$

But I think I remember this failing the parser in TC, and the last two paragraphs seems to favor not supporting this expansion to discourage models with a too high number of parameters.

Reply:
Behavior in TC for reciprocal parameters on more than 2 sublattices seems to be:
- $^0 L_{AB:CD:EF}$ - is stated to not be supported according to the database parser, but equilibrium will still give the expected value of $G_{excess} = y^0_A y^0_B y^1_C y^1_D y^2_E y^2_F * ^0 L_{AB:CD:EF}$
- $^n L_{AB:CD:EF}$ for $n>0$ - is not supported and equilibrium will fail with message: Illegal composition dependency
- $^n L_{AB:C:EF}$ - is supported